website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,159 @@
+
+```
+
+Note, the above command will retrieve all HTTPRoutes and TCPRoutes. TLSRoutes and UDPRoutes are not supported in v0.1.0
+
+If you have any breaking namespaces, your output will something like as follows. If your output is empty, you have no breaking routes and you can skip to the last step.
+
+```
+{
+  "name": "example-breaking-http-route",
+  "namespace": "example-namespace",
+  "kind": "HTTPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+{
+  "name": "example-breaking-tcp-route",
+  "namespace": "a-different-namespace",
+  "kind": "TCPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+```
+
+**3.** Create a ReferencePolicy to allow cross namespace traffic.
+
+Using the above output as a guide, create a [ReferencePolicy](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy) to allow cross namespace traffic. If you've already created a ReferencePolicy you can skip this step.
+
+For example, the above output would require a reference policy that looks follows.
+
+Note, the ReferencePolicy should be created in the same namespace that its allowing traffic to
+
+<CodeBlockConfig filename="referencepolicy.yaml">
+
+
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: ReferencePolicy
+metadata:
+  name: reference-policy
+  namespace: gateway-namespace
+spec:
+  from:
+    - group: gateway.networking.k8s.io
+      kind: HTTPRoute
+      namespace: example-namespace
+    - group: gateway.networking.k8s.io
+      kind: TCPRoute
+      namespace: a-different-namespace
+  to:
+    - group: ""
+      kind: Service
+      name: web-backend
+</CodeBlockConfig>
+
+Edit your ReferencePolicy so all routes are allowed and save into a file called referencepolicy.yaml
+
+Run the following command to apply it to your cluster
+
+```
+kubectl apply -f referencepolicy.yaml
+```
+
+**4.** Upgrade your deployment to v.0.2.0.
+
+```
+kubectl apply --kustomize="github.com/hashicorp/consul-api-gateway/config/crd?ref=0.2.0"
+```
+
+**5.** Helm upgrade consul
+
+```
+helm upgrade --values values.yaml --namespace consul --version <NEW_VERSION> hashicorp/consul <DEPLOYMENT_NAME>
+```
+
+**6.** Refresh gateway deployments
+Because [the gateway class is immutable](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1alpha2.GatewayClass), once you've upgraded your CRD installation, in order to see the effects on prexisting gateways, you will need to delete and recreate any gateways.
```

Comment:
I've got comments I need to add here but I'll do it tomorrow.

Reply:
Most people won't have enough contextual knowledge to understand what it means when we say the gateway class is "immutable." Instead, we can say something like:

```suggestion
A requirement of the Kubernetes Gateway API specification is that the settings of the [Gateway Class](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1alpha2.GatewayClass) are only applied to a gateway when the gateway is created. For that reason, in order to see the effects on preexisting gateways, once you've upgraded your CRD installation,  you will need to delete and recreate any gateways.
```